### ISSUE TITLE
Warn and fix too little memory per core in scylla_setup

### ISSUE THREAD

**Issue by tzach** [scylladb/scylla#2808]
*Installation details*
Scylla version (or git commit hash): 2.0

Scylla will refuse to start when the number of cores is too high compared to the memory, resulting in less than 476 MiB per shard.
This can be identified in scylla_setup suggesting to use a subset of the available cores.

```
You have too little memory for the number of logical processors (CPUs) to run Scylla. To fix, you can reduce the number of processors used.  Reduce the number of processors used [Y/n]? Y
Number of cores to use [receomendation: 4]? 2
Scylla will use 2 cores.
```

**Comment by tzach** [comment]
Suggested by @avikivity in #2803

**Comment by amnonh** [comment]
This is still a problem in situation where scylla_setup runs in non-interactive mode, it would just fail without telling anyone (for example AWS image)

**Comment by tzach** [comment]
> This is still a problem in situation where scylla_setup runs in non-interactive mode, it would just fail without telling anyone (for example AWS image)

Agree. This enhancement request does not cover all use case. Still worthwhile

**Comment by tzach** [comment]
@avikivity looks like you referenced the wrong seastar commit here. Please check.

**Comment by am05mhz** [comment]
is there a way to fix this? the `Configure more memory (--memory option) or decrease shard count (--smp option)` is not very clear. which command to use the parameter --memory or --smp? or where is the config file that sets this options?

also how do you get to see this option?
`You have too little memory for the number of logical processors (CPUs) to run Scylla. To fix, you can reduce the number of processors used.  Reduce the number of processors used [Y/n]? Y
Number of cores to use [receomendation: 4]? 2
Scylla will use 2 cores.`

did not see that when running `scylla_setup`

**Comment by vladzcloudius** [comment]
@am05mhz You can tweak these parameters at any point in time in `/etc/sysconfig/scylla-server`.
Keep in mind, however, that changing the shards count (e.g. using the `--smp` parameter) will trigger "resharding" of existing data.
